strawberry/permission.py

```diff
@@ -56,12 +56,25 @@ def has_permission(
             "Permission classes should override has_permission method"
         )
 
+    def resolve_permission_sync(self, source: Any, info: Info, **kwargs: Any) -> None:
+        if self.has_permission(source, info, **kwargs):
+            return None
+        else:
+            return self.on_unauthorized()
+
+    async def resolve_permission_async(
+        self, source: Any, info: Info, **kwargs: Any
+    ) -> None:
+        if await await_maybe(self.has_permission(source, info, **kwargs)):
+            return None
+        else:
+            return self.on_unauthorized()
+
```

Comment:
Please keep this in the permission extension. The extension has all the logic to handle what to do on failure (e.g. fail silently). The control flow is less clear like this and we need to catch exceptions and rethrow them.

Reply:
The biggest issue is when it comes to composite permissions, the extension probably won't have knowledge of which exact permission failed.  The only solutions I could think of at the time would be a different breaking change (returning an object with information or something similar from has_permission), or having special logic for handling has_permission for the composite permissions (which in my mind seemed confusing?  But might be the way to do it).  Other views/advice eagerly recieved!